test/integration/acceptance/fake_peer_example_test.cpp

```diff
@@ -318,80 +330,98 @@ TEST_F(FakePeerExampleFixture,
-                          })) {
-        got_proposal_from_main_peer_.test_and_set(std::memory_order_relaxed);
-      }
-      HonestBehaviour::processYacMessage(message);
-    }
-
-    const interface::types::HashType &tx_hash_;
-    std::atomic_flag &got_proposal_from_main_peer_;
-  };
-
   // Create the tx:
   const auto tx = complete(
       baseTx(kAdminId).transferAsset(kAdminId, kUserId, kAssetId, "tx1", "1.0"),
       kAdminKeypair);
-  const auto hash = tx.reducedHash();
 
-  std::atomic_flag got_proposal_from_main_peer = ATOMIC_FLAG_INIT;
+  auto state_ready = std::make_shared<std::atomic_bool>(false);
 
-  auto &itf = prepareState(1);
-  fake_peers_.front()->setBehaviour(
-      std::make_shared<CustomBehaviour>(hash, got_proposal_from_main_peer));
+  auto proposal_storage = std::make_shared<fake_peer::ProposalStorage>();
+  proposal_storage->setDefaultProvider(
+      [tx, state_ready](
+          const auto &round) -> fake_peer::OrderingProposalRequestResult {
+        if (!state_ready->load(std::memory_order_relaxed)) {
+          return boost::none;
+        }
+        shared_model::proto::ProtoProposalFactory<
+            validation::AlwaysValidValidator>
+            valid_factory;
+        std::shared_ptr<const shared_model::proto::Proposal> proposal;
+        valid_factory
+            .createProposal(round.block_round,
+                            iroha::time::now(),
+                            std::vector<shared_model::proto::Transaction>({tx}))
+            .match(
+                [&proposal](iroha::expected::Value<std::unique_ptr<
+                                shared_model::interface::Proposal>> &v) {
+                  auto p1 =
+                      std::static_pointer_cast<shared_model::proto::Proposal>(
+                          std::shared_ptr<shared_model::interface::Proposal>(
+                              std::move(v).value));
+                  proposal = std::const_pointer_cast<
+                      const shared_model::proto::Proposal>(std::move(p1));
+                },
+                [](const iroha::expected::Error<std::string> &e) {
+                  FAIL() << "Could not create proposal: " << e.error;
+                });
+        return proposal;
+      });
+
+  auto commit_promise = std::make_shared<std::promise<void>>();
+  std::future<void> commit_future = commit_promise->get_future();
 
+  createFakePeers(1);
+  fake_peers_.front()->setProposalStorage(std::move(proposal_storage));
+
+  auto &itf = prepareState();
+  state_ready->store(true, std::memory_order_relaxed);
   // watch the proposal requests to fake peer
-  bool got_proposal_from_fake_peer = false;
-  fake_peers_.front()->getProposalRequestsObservable().subscribe(
-      [&got_proposal_from_fake_peer](const auto &round) {
-        got_proposal_from_fake_peer = true;
+  itf.getPcsOnCommitObservable()
+      /*
+      .start_with(iroha::synchronizer::SynchronizationEvent{
```

Comment:
Why isn't it deleted?

Reply:
That was a debugging code I forgot to remove, sorry.